shotover-proxy/tests/cassandra_int_tests/mod.rs

```diff
@@ -275,7 +296,77 @@ async fn cluster_multi_rack(#[case] driver: CassandraDriver) {
         shotover_rack3.shutdown_and_then_consume_events(&[]).await;
     }
 
-    cluster::multi_rack::test_topology_task(None).await;
+    let expected_nodes: Vec<SocketAddr> = vec![
+        "172.16.1.2:9042".parse().unwrap(),
+        "172.16.1.3:9042".parse().unwrap(),
+        "172.16.1.4:9042".parse().unwrap(),
+    ];
+    let expected_racks = vec!["rack1", "rack2", "rack3"];
+    cluster::multi_rack::test_topology_task(None, expected_nodes, expected_racks, 128).await;
+}
+
+// This is very slow, only test with one driver
+#[rstest]
+#[case::scylla(Scylla)]
+#[tokio::test(flavor = "multi_thread")]
+async fn cluster_multi_rack_3_per_rack(#[case] driver: CassandraDriver) {
+    let _compose = docker_compose(
+        "tests/test-configs/cassandra/cluster-multi-rack-3-per-rack/docker-compose.yaml",
+    );
+
+    {
+        let shotover_rack1 = shotover_process(
+            "tests/test-configs/cassandra/cluster-multi-rack-3-per-rack/topology_rack1.yaml",
+        )
+        .with_log_name("Rack1")
+        .with_observability_port(9001)
+        .start()
+        .await;
+        let shotover_rack2 = shotover_process(
+            "tests/test-configs/cassandra/cluster-multi-rack-3-per-rack/topology_rack2.yaml",
+        )
+        .with_log_name("Rack2")
+        .with_observability_port(9002)
+        .start()
+        .await;
+        let shotover_rack3 = shotover_process(
+            "tests/test-configs/cassandra/cluster-multi-rack-3-per-rack/topology_rack3.yaml",
+        )
+        .with_log_name("Rack3")
+        .with_observability_port(9003)
+        .start()
+        .await;
+
+        let connection = || async {
+            let mut connection = CassandraConnectionBuilder::new("127.0.0.1", 9042, driver)
+                .build()
+                .await;
+            connection
+                .enable_schema_awaiter("172.16.1.2:9042", None)
+                .await;
+            connection
+        };
+        standard_test_suite_rf3(&connection, driver).await;
+
+        shotover_rack1.shutdown_and_then_consume_events(&[]).await;
+        shotover_rack2.shutdown_and_then_consume_events(&[]).await;
+        shotover_rack3.shutdown_and_then_consume_events(&[]).await;
+    }
+    let expected_nodes: Vec<SocketAddr> = vec![
```

Comment:
Wouldn't be better for this to be a tuple with (IP, Rack)? Or a struct, CassandraNode? It functionally doesn't change anything, just make it more readable. (I know it is a test, so I don't care much anyway :) )

Reply:
yep, good idea!